core/src/main/java/org/elasticsearch/index/query/BoolQueryBuilder.java

```diff
@@ -325,21 +325,17 @@ public static BoolQueryBuilder fromXContent(QueryParseContext parseContext) thro
             } else if (token == XContentParser.Token.START_OBJECT) {
                 switch (currentFieldName) {
                 case MUST:
-                    query = parseContext.parseInnerQueryBuilder();
-                    mustClauses.add(query);
+                    parseContext.parseInnerQueryBuilder().ifPresent(mustClauses::add);
```

Comment:
along with these changes on some of the compound queries, i would like to see the corresponding change in toQuery where we don't have to handle the null lucene query case anymore, as only EmptyQueryBuilder could generate a null lucene query. Also we should test that toQuery never ever returns null in AbstractQueryTestCase.